### PR TITLE
fix wrong column for position of RetInsn when falling off the end of a function

### LINKED ISSUE #784: Wrong column for position of RetInsn when falling off the end of a function
This case

```
public function foo() returns int {
}//whoops
```

will print an error highlighting the comment, not the underscore.

This is due to this line in codeGen.bal:

```
        bir:RetInsn ret = { operand: (), pos: defn.endPos };
```

### PR DESCRIPTION
**Pull request by heshanpadmasiri**
## Purpose
fix wrong column for position of RetInsn when falling off the end of a function
Resolves #784 

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: compiler/modules/front.syntax/parseStmt.bal [hunk 1]
```diff
@@ -9,8 +9,9 @@ function parseStmtBlock(Tokenizer tok) returns StmtBlock|err:Syntax {
             stmts.push(check parseStmt(tok));
         }
         Position endPos = tok.currentEndPos();
+        Position closingPos = tok.currentStartPos();
```
Comment (jclark) [inline compiler/modules/front.syntax/parseStmt.bal]: closeBracePos

File: compiler/modules/front.syntax/ast.bal [hunk 2]
```diff
@@ -73,6 +73,7 @@ public const WILDCARD = ();
 public type StmtBlock record {|
     *PositionFields;
     Stmt[] stmts;
+    Position closingPos;
```
Comment (jclark) [inline compiler/modules/front.syntax/ast.bal]: closeBracePos

**Review by jclark — CHANGES_REQUESTED** [review]

File: compiler/modules/front.syntax/ast.bal (hunk 2 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/ast.bal]: fixed

File: compiler/modules/front.syntax/parseStmt.bal (hunk 1 above)
Comment (heshanpadmasiri) [reply compiler/modules/front.syntax/parseStmt.bal]: fixed

File: compiler/modules/front.syntax/parseStmt.bal [hunk 3]
```diff
@@ -360,7 +361,8 @@ function parseIfElseStmt(Tokenizer tok, Position startPos) returns IfElseStmt|er
             Position blockStartPos = elseIfStmt.ifTrue.startPos;
             StmtBlock? elseIfFalseBlock = elseIfStmt.ifFalse;
             Position blockEndPos = (elseIfFalseBlock ?: elseIfStmt.ifTrue).endPos;
-            ifFalse = { startPos: blockStartPos, endPos: blockEndPos, stmts: [elseIfStmt] };
+            Position closeBracePos = (elseIfFalseBlock ?: elseIfStmt.ifTrue).closeBracePos;
```
Comment (jclark) [inline compiler/modules/front.syntax/parseStmt.bal:364]: This doesn't look good. I created a new issue #794 to clean it up.

**Pull request merged**